### PR TITLE
Add membership validations

### PR DESCRIPTION
**Pull request by kkoehn**
Adds validations to prevent duplicate memberships

### CODE REVIEW

**Comment by kkoehn** [comment]
You are right. The problem is, that groupify uses a Membership with an empty type to signal general membership, if you have additional roles, you get an additional membership with the corresponding type. I think I got it down in a custom uniqueness validation now.

File: app/models/group.rb [hunk 1]
```diff
@@ -17,31 +17,40 @@ def self.create_with_admin(params, user)
...
+  def full_member?(user)
     user.in_group?(self) unless user.in_group?(self, as: 'pending')
   end
 
+  def member?(user)
+    user.in_group?(self, as: 'member')
+  end
+
   def make_admin(user)
+    users.delete(user)
```
Comment (MrSerth) [inline app/models/group.rb]: Do we have a way to remove admin permissions from a member?

Comment (kkoehn) [reply app/models/group.rb]: The there is currently no way to "unadmin" an admin in a group.

File: app/models/group.rb [hunk 2]
```diff
@@ -17,31 +17,40 @@ def self.create_with_admin(params, user)
...
-  def admins
-    User.in_group(self).as(:admin)
-  end
-
   def admin?(user)
     user.in_group?(self, as: 'admin')
   end
 
-  def member?(user)
+  def full_member?(user)
```
[the remark this answers is not in the document]
Comment (kkoehn) [reply app/models/group.rb]: It's good that we don't have to deal with cache invalidation.

`active_member` suggests that there are `inactive` ones as well. I like `confirmed_member` though.

File: app/models/group.rb (hunk 1 above)
Comment (MrSerth) [reply app/models/group.rb]: Alright :)

**Pull request merged**